// @shopify/api-codegen-preset/dist/preset.js
'use strict';

var graphqlCodegen = require('@shopify/graphql-codegen');
var apiConfigs = require('./helpers/api-configs.js');

const preset = {
    buildGeneratesSection: (options) => {
        const apiType = options.presetConfig.apiType;
        const { interfaceExtension, module, presetConfigs } = apiConfigs.apiConfigs[apiType];
        // Use .js extension for import paths - TypeScript resolves these to .ts/.d.ts files
        const typesFile = `${apiConfigs.apiConfigs[apiType].typesFile}.js`;
        return graphqlCodegen.preset.buildGeneratesSection({
            ...options,
            presetConfig: {
                ...presetConfigs,
                importTypes: {
                    namespace: presetConfigs.importTypes.namespace,
                    from: `./${typesFile}`,
                },
                interfaceExtension: ({ queryType, mutationType, }) => interfaceExtension
                    .replace('%%MODULE%%', options.presetConfig.module ?? module)
                    .replace('%%QUERY%%', queryType)
                    .replace('%%MUTATION%%', mutationType),
            },
        });
    },
};

exports.preset = preset;
